scripts/kind-e2e/e2e.sh

```diff
@@ -274,15 +274,36 @@ function cleanup {
 
 ### Main ###
 
-status=$1
-version=$2
-logging=$3
-kubefed=$4
-deploy=$5
-debug=$6
-globalnet=$7
+# Outer loop causes getopts to not-quit on unknown arguments
```

Comment:
1. Why won't we want to quit on unknown params? It's an error.
2. Perhaps it's better to use enhanced getopt which seems more robust, per [1].

[1] https://stackoverflow.com/a/29754866

Reply:
1. Why won't we want to quit on unknown params? It's an error.

IMHO, it's actually a key feature in this case, not a bug. ;) The goal of this work is to make sharing scripts between different CI easier. By ignoring unknown params, different CIs can pass whatever params are relevant to them, and different scripts can catch only the ones they know how to handle.

So for example, the -b option only applies to building images with debug flags, not the e2e.sh script. The build scripts can respond to that flag, while the other scripts can ignore it, and vice versa for the other flags.

Reply:
> 2. Perhaps it's better to use enhanced getopt which seems more robust, per [1].

I thought about that, but went with getopts because it's a Bash built-in. But maybe that's not a good reason, we know the CI box will be Linux. We can try getopts, it'd be good to have long options.

Reply:
I'm not sure I agree, see my other comment about the unknown params from more details.
Also @skitt just told me he's working on improving the Makefile, so peerhaps it's best to coordinate with him on it?